### PR TITLE
Update Datatree html repr to indicate inheritance

### PR DESCRIPTION
**Pull request by TomNicholas**
The HTML version of #9470. ~Haven't written any tests yet~, but in [this notebook](https://gist.github.com/TomNicholas/e1c7fa7a95220fee3aaf9928cc05d499) it looks like it works.

- [x] Closes HTML version of https://github.com/pydata/xarray/issues/9463
- [x] Tests added

### CODE REVIEW

File: xarray/core/formatting_html.py [hunk 1]
```diff
@@ -382,16 +382,40 @@ def summarize_datatree_children(children: Mapping[str, DataTree]) -> str:
...
+inherited_coord_section = partial(
+    _mapping_section,
+    name="Inherited coordinates",
+    details_func=summarize_coords,
+    max_items_collapse=25,
+    expand_option_name="display_expand_coords",
+)
+
+
+def datatree_node_repr(group_title: str, node: DataTree, show_inherited=False) -> str:
```
Comment (shoyer) [inline xarray/core/formatting_html.py:394]: I assume this logic now basically matches that in `datatree_repr`?

**Review by shoyer — APPROVED** [review]
I

File: xarray/core/formatting_html.py (hunk 1 above)
Comment (TomNicholas) [reply xarray/core/formatting_html.py:394]: That's the idea yes

File: xarray/tests/test_formatting_html.py [hunk 2]
```diff
@@ -320,6 +320,25 @@ def test_two_children(
...
+        )
+        with xr.set_options(display_style="html"):
+            html = dt._repr_html_().strip()
+        # checks that the section appears somewhere
+        assert "Inherited coordinates" in html
+
+        # TODO how can we assert that the Inherited coordinates section does not appear in the child group?
+        # with xr.set_options(display_style="html"):
+        #     child_html = dt["a"]._repr_html_().strip()
+        # assert "Inherited coordinates" not in child_html
```
Comment (TomNicholas) [inline xarray/tests/test_formatting_html.py:339]: I'm not really sure how to test this any better. Both the correct and incorrect behaviour would give `"Inherited coordinates"` present somewhere in the long html repr text output.

**Pull request merged**